contracts/governance/staking/StakedToken.sol

```diff
@@ -231,19 +228,22 @@ contract StakedToken is IStakedToken, GamifiedVotingToken {
 
         // 2. Deal with cooldown
         //      If a user is currently in a cooldown period, re-calculate their cooldown timestamp
-        uint256 nextCooldown = getNextCooldownTimestamp(
-            _amount,
-            _msgSender(),
-            balanceOf(_msgSender())
-        );
+        CooldownData memory oldCooldown = stakersCooldowns[_msgSender()];
         //      If we have missed the unstake window, or the user has chosen to exit the cooldown,
         //      then reset the timestamp to 0
         bool exitCooldown = _exitCooldown ||
-            block.timestamp > (nextCooldown + COOLDOWN_SECONDS + UNSTAKE_WINDOW);
+            block.timestamp > (oldCooldown.timestamp + COOLDOWN_SECONDS + UNSTAKE_WINDOW);
         if (exitCooldown) {
             stakersCooldowns[_msgSender()] = CooldownData(0, 0);
+            _exitCooldownPeriod(_msgSender());
+            emit CooldownExited(_msgSender());
         } else {
-            stakersCooldowns[_msgSender()].timestamp = SafeCast.toUint128(nextCooldown);
+            //  Set new percentage so amount being cooled is the same as before the this stake.
+            //  new percentage = old percentage * old staked balance / (old staked balance + new staked amount)
+            uint256 stakedAmountOld = uint256(_balances[_msgSender()].raw);
+            uint256 newPercentage = (oldCooldown.percentage * stakedAmountOld) / (stakedAmountOld + _amount);
+            stakersCooldowns[_msgSender()].percentage = SafeCast.toUint128(newPercentage);
+            _enterCooldownPeriod(_msgSender(), newPercentage);
```

Comment:
Instead of entering a cooldownPeriod here, we should just pass the percentage into `mintRaw`. It will get handled there and save a call